### PR TITLE
Serverless 1 - Intro

### PR DESCRIPTION
**Pull request by joakimkemeny**

### CODE REVIEW

File: _posts/blogg/2017-02-06-serverless-1.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+serverless deploy
+~~~
+
+This will take a couple of minutes since it creates a new stack on AWS and deploys the function. The next deployment will be much faster.
+
+Now open a browser and paste the url from from output into a browser. Congratulations, you’ve just created and deployed your first serverless service and exposed it to the world on an HTTP endpoint.
+
+### What just happened?
+
+Let’s dive into what happened. Serverless created a new stack for you in your AWS account. A stack is like an environment and since you can have multiple stacks you can create one for development, one for testing and one for production or why not one stack for each pull request. Next it deployed the function as a Lambda and configured API Gateway so that when you call the GET endpoint it calls the Lambda.
```
Comment (gustavjorlov) [inline _posts/blogg/2017-02-06-serverless-1.md]: Kanske nämna CloudFormation här.

File: _posts/blogg/2017-02-06-serverless-1.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+If you can see through the weirdness of JavaScript this looks like a normal function that returns a Hello world message, why is this such a big deal? 
+
+As I hope this series will show you this turns out to be a great way to structure your application but what's even better is that when you deploy this function it will cost you absolutely nothing until it's actually used. If your service suddenly gains traction and gets widely popular it will scale automatically so all requests will be served super fast and you will only pay for the actual execution time.
+
+Still not convinced? Let's try it for real to give you an idea of what to expect.
+
+## Getting started
+
+Just as pretty much every cloud provider has its own solution for running containers, many of them are starting go gain support for serverless architectures. I’ll get back to this in a later post but for now I’ll be focusing on AWS' implementation called Lambda just because it’s the most mature solution and it also happens to be the one that I’ve been working with for the last year.
```
Comment (Changis) [inline _posts/blogg/2017-02-06-serverless-1.md]: go gain --> to gain ?

File: _posts/blogg/2017-02-06-serverless-1.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+~~~ shell
+serverless create --template aws-nodejs
+~~~
+
+This will create a normal Node.js project and install its dependencies. The interesting parts can be found in `handler.js` that contains the actual function that we want to run and the file `serverless.yml` that contains the configuration that tells AWS how to deploy the function and how to trigger it. How should the function be triggered?
+
+### Events
+
+Even if functions are the heros of serverless architectures they still need to be triggered somehow. The trigger can be a lot of things; a message on a queue, an event from an IoT device or maybe another function. I'll get back to these in a later post but for now we are going to use a normal HTTP request to trigger the function.
```
Comment (Changis) [inline _posts/blogg/2017-02-06-serverless-1.md]: heroes*

File: _posts/blogg/2017-02-06-serverless-1.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+Next let’s deploy and run this.
+
+~~~ shell
+serverless deploy
+~~~
+
+This will take a couple of minutes since it creates a new stack on AWS and deploys the function. The next deployment will be much faster.
+
+Now open a browser and paste the url from from output into a browser. Congratulations, you’ve just created and deployed your first serverless service and exposed it to the world on an HTTP endpoint.
```
Comment (Changis) [inline _posts/blogg/2017-02-06-serverless-1.md]: from x2

File: _posts/blogg/2017-02-06-serverless-1.md (hunk 2 above)
Comment (joakimkemeny) [inline _posts/blogg/2017-02-06-serverless-1.md]: Done

File: _posts/blogg/2017-02-06-serverless-1.md (hunk 3 above)
Comment (joakimkemeny) [inline _posts/blogg/2017-02-06-serverless-1.md]: Done

File: _posts/blogg/2017-02-06-serverless-1.md (hunk 4 above)
Comment (joakimkemeny) [inline _posts/blogg/2017-02-06-serverless-1.md]: Done

File: _posts/blogg/2017-02-06-serverless-1.md (hunk 1 above)
Comment (joakimkemeny) [inline _posts/blogg/2017-02-06-serverless-1.md]: Kanske något sånt här:

>  ...It also created the neccessary roles for the services to comunicate with each other. 
>
>Serverless doesn't have a magic wand to do this, instead it uses CloudFormation which is the solution that AWS provides to automate deployments. CloudFormation is quite complicated though so Serverless makes this much easier.

Comment (gustavjorlov) [inline _posts/blogg/2017-02-06-serverless-1.md]: Pedagogiskt!

File: _posts/blogg/2017-02-06-serverless-1.md [hunk 5]
```diff
@@ -96,11 +96,13 @@ serverless deploy
...
+Now open a browser and paste the url from output into a browser. Congratulations, you’ve just created and deployed your first serverless service and exposed it to the world on an HTTP endpoint.
 
 ### What just happened?
 
-Let’s dive into what happened. Serverless created a new stack for you in your AWS account. A stack is like an environment and since you can have multiple stacks you can create one for development, one for testing and one for production or why not one stack for each pull request. Next it deployed the function as a Lambda and configured API Gateway so that when you call the GET endpoint it calls the Lambda. 
+Let’s dive into what happened. Serverless created a new stack for you in your AWS account. A stack is like an environment and since you can have multiple stacks you can create one for development, one for testing and one for production or why not one stack for each pull request. Next it deployed the function as a Lambda and configured API Gateway so that when you call the GET endpoint it calls the Lambda.  It also created the neccessary roles for the services to comunicate with each other.
```
Comment (gustavjorlov) [inline _posts/blogg/2017-02-06-serverless-1.md]: -> communicate

**Pull request merged**